src/CG/CG_elem.cpp

```diff
@@ -2511,11 +2531,10 @@ bool intersectPointSegment( array3D const &P, array3D const &P1, array3D const &
 {
     assert( validSegment(P1,P2) );
 
-    if(distancePointSegment(P, P1, P2) > distanceTolerance){
-        return false;
-    }
+    array3D xP     = projectPointSegment( P, P1, P2 );
```

Comment:
In the old implementation, this point was considered intersecting the segment:

![point_segment_intersection](https://github.com/optimad/bitpit/assets/7443174/863a509b-c689-4bc9-ba9a-1ad810ea59d6)

Reply:
But you may be right, distancePointSegment seems the function that contains the bug.

Reply:
I think @edoardolombardi is right, but I still see different (and more correct) results in immerflow with the new implementation. I have to investigate more to see what's going on.

Reply:
To have an exact match between the two implementations, the updated one should check for "<=" rather than for just "<". However, I don't think this will make any difference for your test case.

Reply:
I'm studying a case where a plane is almost intersecting a segment. The distance of the projection point P from the segment edge P2 (see above figure) is of order 10^-20. In that case, the order of operations plays a significant role. So, bitpit and an algorithm I've developed to compute intersections were giving me contradictive results. That's why I got this false alarm.

Since bitpit is not wrong, I suggest not to merge this branch and cancel the pull-request. I can continue my work by changing the tolerance used in bitpit to compare distances.

Reply:
(Please also replace the < with a <= to make the results as close as possible to the previous implementation.)

Reply:
Thanks! I did what you asked.